crates/pathfinder/src/rpc/v02/method/add_deploy_account_transaction.rs

```diff
@@ -0,0 +1,122 @@
+use anyhow::Context;
+use serde_with::serde_as;
+
+use crate::rpc::serde::{FeeAsHexStr, TransactionVersionAsHexStr};
+use crate::{
+    core::{
+        CallParam, ClassHash, ContractAddress, ContractAddressSalt, Fee, StarknetTransactionHash,
+        TransactionNonce, TransactionSignatureElem, TransactionVersion,
+    },
+    rpc::v02::RpcContext,
+    sequencer::ClientApi,
+};
+
+const fn transaction_version_zero() -> TransactionVersion {
+    TransactionVersion(web3::types::H256::zero())
+}
+
+#[serde_as]
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+#[serde(deny_unknown_fields)]
+pub struct AddDeployAccountTransactionInput {
```

Comment:
Should I re-organize code somehow to avoid using too much of `crate::core::*`?

Reply:
Iiuc - there's no need to do so. Even though starknet essentially uses `felt` for everything we introduced the newtypes in `core` around it to distinguish based on purpose/meaning.

Reply:
Noted, thanks.